compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/DecodeV1.hs

```diff
@@ -24,14 +24,18 @@ import qualified Data.Text.Lazy as TL
 import qualified Data.Vector as V
 import qualified Proto3.Suite as Proto
 
--- internal functions that *implement* decoding
-type DecodeImpl = ReaderT PackageRefCtx Decode
-type MonadDecode = MonadError Error
 
-decodeImpl :: MonadDecode m => Decode a -> m a
-decodeImpl = either throwError pure
+newtype DecodeEnv = DecodeEnv
+    { internedStrings :: V.Vector T.Text
```

Comment:
This looks like a new addition here? Or does this replace the `PackageRefCtx`?

Reply:
It replaces the `PackageRefCtx`. I somehow prefer passing data rather than functions around when possible.